src/EventListener/ReadRateLimitAnnotationListener.php

```diff
@@ -98,4 +125,16 @@ public static function getSubscribedEvents(): array
             ControllerEvent::class => 'onKernelController',
         ];
     }
+
+    public function extractQueryName($query): string
+    {
+        $t = Parser::parse($query);
+        /** @var OperationDefinitionNode $item */
+        foreach ($t->definitions->getIterator() as $item) {
+            /** @var FieldNode $item2 */
+            return (string) $item->selectionSet->selections[0]->name->value;
+        }
```

Comment:
Have you tried to "profile" this ?
I don't know but it may have an impact on perfs ?
(I usually think about when I see a foreach for each calls / mutation 🤔 )

Reply:
I has a cost of 5ms on the tests I have done

Reply:
There are already around 40ms for rate limit get and storage 🤷‍♀️

Reply:
after many search I find nothing better than using Parser and move over the tree. But in fact we can remove the foreach because we only loop one time